v2/common/src/main/java/com/google/cloud/teleport/v2/utils/DataStreamClient.java

```diff
@@ -458,4 +535,72 @@ public StandardSQLTypeName convertMysqlToBigQueryColumnType(MysqlColumn column)
       return StandardSQLTypeName.STRING;
     }
   }
+
+  public StandardSQLTypeName convertPostgresqlToBigQueryColumnType(PostgresqlColumn column) {
+    String dataType = column.getDataType().toUpperCase();
+
+    switch (dataType) {
+      case "BOOLEAN":
+        return StandardSQLTypeName.BOOL;
+      case "BLOB":
+      case "VARCHAR":
+      case "CHAR":
+      case "CHARACTER":
+      case "CHARACTER VARYING":
+      case "TINYTEXT":
+      case "TEXT":
+      case "MEDIUMTEXT":
+      case "LONGTEXT":
+      case "UUID":
+      case "XML":
+        return StandardSQLTypeName.STRING;
+      case "TINYINT":
+      case "SMALLINT":
+      case "MEDIUMINT":
+      case "INT":
+      case "INTEGER":
+      case "SMALLSERIAL":
+      case "SERIAL":
+      case "BIGINT":
+        return StandardSQLTypeName.INT64;
+      case "REAL":
+      case "DOUBLE PRECISION":
+        return StandardSQLTypeName.FLOAT64;
+      case "DECIMAL":
+      case "NUMERIC":
+        return StandardSQLTypeName.BIGNUMERIC;
+      case "BIT":
+      case "BIT VARYING":
+      case "BYTEA":
+        return StandardSQLTypeName.BYTES;
+      case "DATETIME":
+        return StandardSQLTypeName.TIMESTAMP;
+      case "DATE":
+        return StandardSQLTypeName.DATE;
+      case "JSON":
+      case "JSONB":
+        return StandardSQLTypeName.JSON;
+        // (naveronen) - i'm setting this a STRING for now, but some customers might
+        // need a
+        // different
+        // solution. once we encounter such cases, we might need to adjust this
+      case "SET":
+      case "ENUM":
+        return StandardSQLTypeName.STRING;
+      default:
+    }
+
+    if (TIMESTAMP_PATTERN.matcher(dataType).matches()) {
+      return StandardSQLTypeName.TIMESTAMP;
+    } else if (TIMESTAMP_WITHOUT_TIMEZONE_PATTERN.matcher(dataType).matches()) {
+      return StandardSQLTypeName.TIMESTAMP;
+    } else if (TIMESTAMP_WITH_TIMEZONE_PATTERN.matcher(dataType).matches()) {
+      return StandardSQLTypeName.TIMESTAMP; // TODO: what type do we want here?
+    } else if (TIMESTAMP_WITH_LOCAL_TIMEZONE_PATTERN.matcher(dataType).matches()) {
+      return StandardSQLTypeName.TIMESTAMP; // TODO: what type do we want here?
```

Comment:
I'm thinking Postgres doesn't have this one. If the timezone is included with it though, then it should be `StandardSQLTypeName.TIMESTAMP`, otherwise `StandardSQLTypeName.DATETIME`

Reply:
the default `TIMESTAMP` from postgres (with it's value from `CURRENT_TIMESTAMP`) had it's `INFORMATION_SCHEMA` data type entry stored as `timestamp without time zone`, but yeah let's test it out